File: 백준/Silver/7795. 먹을 것인가 먹힐 것인가/먹을 것인가 먹힐 것인가.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    int testCase; cin >> testCase;
    while(testCase--) {
        int n, m;
        cin >> n >> m;

        vector<unsigned long long> a;
        vector<unsigned long long> b;


        for (int i = 0; i < n; i++) {
            unsigned long long tmp; cin >> tmp;
            a.emplace_back(tmp);
        }

        for (int i = 0; i < m; i++) {
            unsigned long long tmp; cin >> tmp;
            b.emplace_back(tmp);
        }   

        unsigned long long answer = 0;

        sort(a.begin(), a.end(), greater<unsigned long long>());
        sort(b.begin(), b.end(), greater<unsigned long long>());

        int i = 0;
        int j = 0;

        while (true) {
            if (a[i] > b[j]) {
                answer += m-j;
                i++;
            }
            else {
                j++;
            }
            if ( i >= n || j >= m ) break;
        }

        cout << answer << '\n';
    }
}

File: 백준/Silver/7795. 먹을 것인가 먹힐 것인가/먹을 것인가 먹힐 것인가.py
import sys

testCase = int(sys.stdin.readline())

for i in range(testCase):
    n, m = map(int, sys.stdin.readline().split())  

    a = list(map(int, sys.stdin.readline().split()))
    b = list(map(int, sys.stdin.readline().split()))

    a.sort(reverse=True)
    b.sort(reverse=True)

    i = 0
    j = 0

    answer = 0

    while (i < n and j < m):
        if (a[i] > b[j]):
            answer += m-j
            i += 1
        else:
            j += 1

    print(answer)